### DISCUSSION TITLE
Create multiple form by input value

### DISCUSSION THREAD

**Discussion by slafragola93** [symfony/symfony#47796] [Q&A]
Hi guys,

I need to create some forms dynamically. I explain.

I make a form with the "form builder", now i need that my application will create the number of form that the user have insert inside an input. 

So for exemple, if the user put 4 inside 'nb_colli', i want create a loop that will generate 4 forms. But i want do this dynamically, so when the user put the number, my script js will get the number and will generate the forms. 

Obviously, this thing is not possible, because all the fields are already rendered 1 time and is not possible to do another. 

How can i provide to this problem? I have applicate this logic:

**OrderDetailType**
`
public function buildForm(FormBuilderInterface $builder, array $options): void
    {
     $builder
            ->add('id_order',HiddenType::class)
            ->add('nb_colli', NumberType::class,  [
                'required' => true,
                'mapped' => false,
                'label' => 'Colli'
            ])
            ->add('height', NumberType::class,  [
                'required' => true,
                'label' => 'Altezza',
            ])
           ->add('height', NumberType::class,  [
                  'required' => true,
                  'label' => 'Altezza',
              ])
            ->add('depth', NumberType::class,  [
                'required' => true,
                'label' => 'Profondità'
            ])
            ->add('length', NumberType::class,  [
                'required' => true,
                'label' => 'Lunghezza'
            ])
            ->add('value', NumberType::class,  [
                'required' => true,
                'label' => 'Valore'
            ])
            ->add('save', SubmitType::class, [
                'label' => 'Salva'
            ])
        ;
    }`
    
    This is my JS
    `$("#nb_colli").on("keyup", function () {
		$('#order_detail').children().empty();
		let nb_colli_manuale = $('#order_detail_nb_colli').val();
		var final_html = "";
		if (nb_colli_manuale > 15) {
			final_html = '<h1>Ridurre i colli da inserire</h1>';
		} else {
			for (let i = 0; i < nb_colli_manuale; i++) {
				final_html += "<tr>";
				final_html += "{{ form_row(pack_details_form.height) }}";
				final_html += "{{ form_row(pack_details_form.length) }}";
				final_html += "{{ form_row(pack_details_form.depth) }}";
				final_html += "{{ form_row(pack_details_form.weight) }}";
				final_html += "{{ form_row(pack_details_form.value) }}";
				final_html += "</tr>";
			}
			final_html += "<tr><td>{{ form_row(pack_details_form.save) }}</td></tr>";
		}
			$('#order_detail').append(final_html);
	})`

**Comment by MatTheCat** [comment]
Seems like a basic use case for [the CollectionType field](https://symfony.com/doc/current/reference/forms/types/collection.html)?
